### PR TITLE
FileSelectorButton doesn't fire event if disabled

### LINKED ISSUE #10548: FileSelectorButton still fires events even when disabled
The `qx.ui.form.FileSelectorButton` still fires events like `changeFileSelection` even when `setEnabled(false)` has been called on it.

### PR DESCRIPTION
**Pull request by goldim**
Fixed #10548

### CODE REVIEW

**Comment by goldim** [comment]
Not sure if do the same for "tap" event. The "execute" event is not fired I can't catch it.

**Review by hkollmann — APPROVED** [review]

**Comment by hkollmann** [comment]
@goldim : BTW you can use es6 syntax here!

**Comment by goldim** [comment]
@hkollmann Exactly! Thanks.

**Review by oetiker — APPROVED** [review]

**Review by derrell — CHANGES_REQUESTED** [review]

File: source/class/qx/ui/form/FileSelectorButton.js [hunk 1]
```diff
@@ -147,6 +147,12 @@ qx.Class.define("qx.ui.form.FileSelectorButton", {
         { id: id }
       ));
 
+      input.addListener("click", e => {
```
[the remark this answers is not in the document]
Comment (goldim) [reply source/class/qx/ui/form/FileSelectorButton.js:150]: @derrell any suggestion?

[the remark this answers is not in the document]
Comment (derrell) [reply source/class/qx/ui/form/FileSelectorButton.js:150]: I guess my question is why it's even getting to here, on "click"? It's a `qx.html.Input`, and looking in that source code, it has this:

```
    setEnabled(value) {
      this.__enabled = value;

      this.setAttribute("disabled", value === false);
      ...
```

So why is a disabled input able to generate events? Maybe the value passed to `setEnabled` isn't a strict boolean, `false`? Maybe the above line should be:

```
      this.setAttribute("disabled", ! value);
```

**Review by derrell — CHANGES_REQUESTED** [review]

**Review by derrell — APPROVED** [review]
This looks like it fixes the bug in `FileSelectorButton`. I fear it still leaves a bug in anything else that uses `Input` since I don't understand how a disabled element can fire an event...

**Comment by goldim** [comment]
@derrell there is no 100% chance to avoid an error. There is a button which includes an input and I don't see how they are linked talking about enabled property. If the button is a container for input then I think changing a button property will lead to changing the similar input property. And what errors could happen looking at what I changed. Ive just bound enabled prop of two widgets.

**Comment by derrell** [comment]
@goldim In qx.html.Input, this line:

```
this.setAttribute("disabled", value === false);
```

is setting the `disabled` attribute of the DOM element. https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled

Since a disabled DOM element can't be focused, etc., I would expect that it couldn't generate an event. Thus, my assumption is that line of code isn't doing what we expect, possibly because the argument to `setEnabled` is something falsy like `0`, but not `false` itself, so the element isn't being disabled.

**Comment by derrell** [comment]
I commented, but deleted my comment. You are correct. Your change looks good.

**Review by johnspackman — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Comment by hkollmann** [comment]
Thanks @goldim

**Pull request merged**